modules/core/include/opencv2/core/eigen.hpp

```diff
@@ -287,6 +287,16 @@ void cv2eigen( const Mat& src,
     }
 }
 
+template<typename _Tp>  static inline
+void cv2eigen( const Mat& src,
+               Eigen::Matrix<_Tp, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>& dst )
+{
+    dst.resize(src.rows, src.cols);
```

Comment:
check for `src.dims == 2`:
`CV_CheckEQ(src.dims, 2, "");`

Reply:
Fixed